targetFunctions/+domainsTF/customLayers.m

```diff
@@ -17,11 +17,24 @@
     % Extract individual parameters from problemStruct
     [numberOfContrasts, geometry, contrastBackgroundIndices, contrastQzshiftIndices,...
      contrastScalefactorIndices, contrastBulkInIndices, contrastBulkOutIndices,...
-     contrastResolutionIndices, contrastDomainRatioIndices, backgroundParamArray,...
+     contrastResolutionParamIndices, contrastDomainRatioIndices, backgroundParamArray,...
      qzshiftArray, scalefactorArray, bulkInArray, bulkOutArray, resolutionParamArray,...
      domainRatioArray, dataPresent, nParams, params, ~, resample,...
      contrastBackgroundActions, cCustFiles, useImaginary] = extractProblemParams(problemStruct);
     
+    checkIndices(contrastBackgroundIndices, contrastQzshiftIndices,...
```

Comment:
Is it possible to do this check earlier maybe in `parseClassToStructs`? This should reduce duplication and any chance of affecting performance. The function can take problemStruct as input also

Reply:
I was cautious owing to the fact that the reflectivity calculation is performed numerous times within the minimisers, with changes to problemStruct. Having looked through carefully though, it seems like the changes made in unpackParams cannot affect the length of the parameter arrays, so we should be ok. Let me know if you can think of anything that might be affected though.